lib/rules/no-restricted-class.js

```diff
@@ -5,16 +5,25 @@
 'use strict'
 
 const utils = require('../utils')
+const regexp = require('../utils/regexp')
 
 /**
  * Report a forbidden class
  * @param {string} className
  * @param {*} node
  * @param {RuleContext} context
- * @param {Set<string>} forbiddenClasses
+ * @param {Array<string>} forbiddenClasses
  */
 const reportForbiddenClass = (className, node, context, forbiddenClasses) => {
-  if (forbiddenClasses.has(className)) {
+  const isForbidden = forbiddenClasses.some((cl) => {
+    if (regexp.isRegExp(cl)) {
+      const re = regexp.toRegExp(cl)
+      return re.test(className)
+    } else {
```

Comment:
Doing this in every iteration seems unnecessarily expensive. It should be done once when parsing the options (in the `create` function).

Probably `forbiddenClasses` should stay a `Set`, so `forbiddenClasses.has(className)` still is a constant-time lookup, and then a new `forbiddenClassesRegexps` can be an array that you can check additionally.